Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.Common/Providers/IGraphUtility.cs

```diff
@@ -20,83 +20,76 @@ public interface IGraphUtility
+        Task<HttpResponseMessage> RegisterWorkforceIntegrationAsync(Models.RequestModels.WorkforceIntegration workforceIntegration, GraphConfigurationDetails graphConfigurationDetails);
 
         /// <summary>
         /// Fetch user details for shifts using graph api tokens.
         /// </summary>
-        /// <param name="accessToken">Access Token.</param>
+        /// <param name="graphConfigurationDetails">Graph configuration details.</param>
         /// <returns>The shift user.</returns>
-        Task<List<ShiftUser>> FetchShiftUserDetailsAsync(
-            string accessToken);
+        Task<List<ShiftUser>> FetchShiftUserDetailsAsync(GraphConfigurationDetails graphConfigurationDetails);
 
         /// <summary>
         /// Share the schedule between the given date range.
         /// </summary>
-        /// <param name="accessToken">The MS GraphAPI token.</param>
+        /// <param name="graphConfigurationDetails">The graph configuration details.</param>
         /// <param name="teamId">The id of the team whos schedule we want to share.</param>
         /// <param name="startDateTime">The start time we want to share from.</param>
         /// <param name="endDateTime">The end time we want to share until.</param>
         /// <param name="notifyTeam">Whether we want to notify the team or not.</param>
         /// <returns>Http response message.</returns>
-        Task<HttpResponseMessage> ShareSchedule(string accessToken, string teamId, DateTime startDateTime, DateTime endDateTime, bool notifyTeam);
+        Task<HttpResponseMessage> ShareSchedule(GraphConfigurationDetails graphConfigurationDetails, string teamId, DateTime startDateTime, DateTime endDateTime, bool notifyTeam);
 
         /// <summary>
         /// Fetch user details for shifts using graph api tokens.
         /// </summary>
-        /// <param name="accessToken">Access Token.</param>
+        /// <param name="graphConfigurationDetails">The graph configuration details.</param>
         /// <returns>The shift user.</returns>
-        Task<List<ShiftTeams>> FetchShiftTeamDetailsAsync(
-            string accessToken);
+        Task<List<ShiftTeams>> FetchShiftTeamDetailsAsync(GraphConfigurationDetails graphConfigurationDetails);
+
+        /// <summary>
+        /// Sends a graph request.
+        /// </summary>
+        /// <param name="graphConfigurationDetails">The graph configuration details.</param>
+        /// <param name="httpClient">The http client.</param>
+        /// <param name="httpMethod">The http method.</param>
+        /// <param name="requestUrl">The request URI.</param>
+        /// <param name="requestString">The request string to add as the request content.</param>
+        /// <returns>A HttpResponseMessage object.</returns>
+        Task<HttpResponseMessage> SendGraphHttpRequest(GraphConfigurationDetails graphConfigurationDetails, HttpClient httpClient, HttpMethod httpMethod, string requestUrl, string requestString = null);
```

Comment:
nit: the class is GraphUtility so you don't really need Graph in the method name, it is kind of a given?